### PR TITLE
feat: Adding analytics tracking to the secondary Publication filter

### PR DESCRIPTION
**Pull request by carolinechoi**
## feat: Adding analytics to the Publication filter 

- #4953
- #4954 

## Changes

- add FILTER_SELECT_PUBLICATION
- modify WMG_SHARE_CLICKED functionality (expanded to take publication filter)
- modify WMG_SHARE_LOADED functionality ("")

## Testing steps

- need to test in rdev for Amanda

### CODE REVIEW

File: frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveExport/csvUtils.ts [hunk 1]
```diff
@@ -58,7 +65,7 @@ export function csvHeaders({
   output.push([
     `# Link Generated: ${generateAndCopyShareUrl({
       compare,
-      filters: selectedFilters,
+      filters: selectedFilters, // Caroline,  check me
```
Comment (seve) [inline frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveExport/csvUtils.ts:68]: Caroline, delete me

File: frontend/src/views/WheresMyGene/components/GeneSearchBar/components/ShareButton/index.tsx [hunk 2]
```diff
@@ -78,6 +82,7 @@ export default function ShareButton(): JSX.Element {
       const loadedState = loadStateFromQueryParams(
         params,
         selectedFilters,
+        // selectedPublicationFilter, // CAROLINE COME BACK 2 THIS
```
Comment (seve) [inline frontend/src/views/WheresMyGene/components/GeneSearchBar/components/ShareButton/index.tsx:85]: did you ever make it back?

**Review by seve — CHANGES_REQUESTED** [review]

**Review by seve — APPROVED** [review]

File: frontend/src/common/queries/wheresMyGene.ts [hunk 3]
```diff
@@ -499,13 +501,32 @@ export function useFilterDimensions(version: 1 | 2 = 1): {
         return [
           {
             id: collection.id,
-            name: collection.summaryCitation || "",
+            name: collection.summaryCitation || "No Publication",
```
Comment (joyceyan) [inline frontend/src/common/queries/wheresMyGene.ts:504]: might be nice to have the string `"No Publication"` set as a variable somewhere so you can reference it throughout this file

File: frontend/src/common/queries/wheresMyGene.ts [hunk 4]
```diff
@@ -531,6 +556,8 @@ export function useFilterDimensions(version: 1 | 2 = 1): {
       );
     }
 
+    console.log(intersect);
```
Comment (joyceyan) [inline frontend/src/common/queries/wheresMyGene.ts:559]: generally we want to avoid merging print statements into `main`

**Review by joyceyan — APPROVED** [review]
for the test failure, it looks like it could be flaky to me. i'd re-run the test a few times before spending too much time debugging the failure

File: frontend/src/common/queries/wheresMyGene.ts (hunk 4 above)
Comment (prathapsridharan) [reply frontend/src/common/queries/wheresMyGene.ts:559]: +1

File: frontend/src/common/queries/wheresMyGene.ts (hunk 3 above)
Comment (prathapsridharan) [reply frontend/src/common/queries/wheresMyGene.ts:504]: +1

Comment (carolinechoi) [reply frontend/src/common/queries/wheresMyGene.ts:504]: Agreed! That change is reflected in the "No Publication" PR. Here: https://github.com/chanzuckerberg/single-cell-data-portal/pull/5047

File: frontend/src/common/queries/wheresMyGene.ts (hunk 4 above)
Comment (carolinechoi) [reply frontend/src/common/queries/wheresMyGene.ts:559]: Oops thank you for catching!!

**Pull request merged**